File: exams/gavilanez/u3/ExamP3(2)/ec_edu_espe_examp3_model/InsertionSort.py
from ec_edu_espe_examp3_controller.SortingStrategy import SortingStrategy

class InsertionSort(SortingStrategy):
    def sort(self, numbers):
        sorted_numbers = numbers[:]
        for i in range(1, len(sorted_numbers)):
            key = sorted_numbers[i]
            j = i-1
            while j >= 0 and key < sorted_numbers[j]:
                sorted_numbers[j + 1] = sorted_numbers[j]
                j -= 1
            sorted_numbers[j + 1] = key
        return sorted_numbers

File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJavaScript/controller/InsertionSort.js
export class InsertionSort {
    sort(numbers) {
        const n = numbers.length;
        for (let i = 1; i < n; ++i) {
            let key = numbers[i];
            let j = i - 1;

            while (j >= 0 && numbers[j] > key) {
                numbers[j + 1] = numbers[j];
                j = j - 1;
            }
            numbers[j + 1] = key;
        }
    }

    getStrategyName() {
        return "Insertion Sort";
    }
}
